webapp/blog/views.py

```diff
@@ -117,8 +117,14 @@ def feed():
     return flask.Response(right_title, mimetype="text/xml")
 
 
+@blog.route(
+    '/<regex("[0-9]{4}"):year>/<regex("[0-9]{2}"):month>/'
+    '<regex("[0-9]{2}"):day>/<slug>'
+)
+@blog.route('/<regex("[0-9]{4}"):year>/<regex("[0-9]{2}"):month>/<slug>')
+@blog.route('/<regex("[0-9]{4}"):year>/<slug>')
```

Comment:
Can this go on it's own view that redirects directly to the slug. This will avoid a useless api call or logic treatement.

```python

@blog.route(
    '/<regex("[0-9]{4}"):year>/<regex("[0-9]{2}"):month>/'
    '<regex("[0-9]{2}"):day>/<slug>'
)
@blog.route('/<regex("[0-9]{4}"):year>/<regex("[0-9]{2}"):month>/<slug>')
@blog.route('/<regex("[0-9]{4}"):year>/<slug>')
def redirect_article(slug, year=None, month=None, day=None):
    return flask.redirect(flask.url_for(".article"), slug="{slug}".format(**locals()))
```

Reply:
And you can add test to make sure the redirection redirects :smile:

Reply:
:+1: Yep fair point. I'll do that.